lib/train-winrm/connection.rb

```diff
@@ -107,15 +107,47 @@ def file_via_connection(path)
         Train::File::Remote::Windows.new(self, path)
       end
 
-      def run_command_via_connection(command, &data_handler)
+      def run_command_via_connection(command, opts = {}, &data_handler)
         return if command.nil?
 
         logger.debug("[WinRM] #{self} (#{command})")
         out = ""
+        response = nil
+        timeout = opts[:timeout]&.to_i
+
+        # Run the command in a thread, to support timing out the command
```

Comment:
We now always run the command in a thread. Possibly, if the timeout option is not set, the command could run using the previous code, without using a thread. That would simplify execution, but would require some duplicated lines of code